v2/src/components/webJsInjector/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { PropsWithChildren } from "react";
+import { recursiveMap } from "../utils";
+import getURI from "../api/webJs";
+import { MOCK_ENABLED } from "../constants";
+
+type Uri = {
+  dateprovider: string;
+  emailpassword: string;
+  emailverification: string;
+  multifactorauth: string;
+  multitenancy: string;
+  passwordless: string;
+  session: string;
+  supertokens: string;
+  thirdparty: string;
+  totp: string;
+  userroles: string;
+  website: string;
+};
+
+type State = {
+  uri: Uri | undefined;
+};
+export default class WebJsInjector extends React.PureComponent<
+  PropsWithChildren<{}>,
+  State
+> {
+  isUnmounting = false;
+
+  constructor(props: PropsWithChildren<{}>) {
+    super(props);
+    this.state = {
+      uri: undefined,
+    };
+  }
+
+  replaceUrisByKeys(value: string) {
+    if (!this.state.uri) {
+      // replace all webjs mentions with empty string;
+      return value.replace(/\^\{jsdeliver_webjs_[^}]+\}/g, "");
+    }
+
+    const uri = this.state.uri;
+    return Object.keys(uri).reduce((acc, key) => {
+      acc = acc.replace(
+        new RegExp(`\\^\\{jsdeliver_webjs_${key}\\}`, "g"),
+        uri[key as keyof State["uri"]]
+      );
+      return acc;
+    }, value);
+  }
+
+  render() {
+    let result = recursiveMap(this.props.children, (c: any) => {
+      if (typeof c === "string") {
+        c = this.replaceUrisByKeys(c);
+      }
+      return c;
+    });
+
+    return result;
+  }
+
+  async componentDidMount() {
```

Comment:
this PR looks good. But i think the API spec will change based on my comment in the other pr, so this may require changes?